### DISCUSSION TITLE
Modrinth Points system and "Pay what you want" feature for projects

### DISCUSSION THREAD

**Discussion by brawaru** [modrinth/.github#275] [Feature requests]
> [!NOTE]
> This is more like a draft for exploration right now and so that I don't forget. I might further elaborate on it when feel better and have more energy, as well as if there's interest, but you are welcome to chip in for discussion already.

## Problem

For some creators modding has grown out of just a hobby and requires the same labour that a real job does. Currently creators rely solely on ad revenue to pay back for their labour or that someone might notice link to their Patreon/Ko-fi/BMAC page and consider donating. This is very unrewarding.

At the same time, Modrinth is now in search for ways to sustain its development, and one of the current ways is introduction of a subscription to Modrinth, which regular people may not consider because it doesn't provide any value to them.

## Goal

Encourage donations to creators on the platform and to further sustain Modrinth operations and development while avoiding breaking EULA, payment platform agreements, or application stores rules.

### Modrinth Points

Modrinth Points would be a special medium on Modrinth platform that can be bought, received for ads displayed for published projects, transferred to other users, and exchanged for real world currency. It intends to replace the current ad revenue in USD by converting it to points.

- 1 point is valued 0.0009 USD. Withdrawing 1000 points gives creator 0.9 USD (including fees?).
- Points are sold in packages:
  - 2500 (2.49 USD · profit 0.24 USD)
  - 5000 (4.99 USD · profit 0.49 USD)
  - 10000 (9.99 USD · profit 0.99 USD)
  - 50000 (49.99 USD · profit 4.99 USD)
- (Area to explore:) Discount to buying points by subscribing to Modrinth or free points to spend every month of subscription.

Values are entirely theoretical and may be different if this proposal was to be implemented, especially if proposal is to include payment processors fees. The point is - Modrinth will earn from purchases of points.

Inspiration: Twitch Bits, Telegram Stars.

### ‘Pay what you want’ (points exchange)

Creators on the platform will gain the ability to replace standard "download" buttons with window encouraging to pay a specific amount of points before downloading. The user is free to dismiss the window and pay nothing, or adjust the amount of points they want to pay.

The user does not in reality pay for a project, but rather just transfers their points to the creator. It's not a substitute for a real payment either, because user has to purchase the points bundle, they cannot buy a specific amount of points.

### Users perception

- A lot of people hate Minecraft Marketplace because 1) part of Minecraft community feels like they're entitled to free content, 2) people hate micro-transactions as they involve FOMO. We should make it clear that purchasing points is entirely optional and that by using this system you are supporting the creators on the platform to create more content and Modrinth operations to provide platform for these creators. It's not us wanting to get rich, it's us wanting to make modding rewarding and our mission sustainable.

- There might be misconception that this system violates Minecraft EULA, specifically the part about ‘making money from mods’. It must be conveyed clearly that this system is entirely contained within Modrinth platform and thus users just exchange points between each other on the platform and never actually pay directly for mods or other creations. The ‘pay what you want’ is merely a reminder to use points and does not paywall the content.

**Comment by ToBinio** [comment]
> a subscription to Modrinth, which regular people may not consider because it doesn't provide any value to them.

in which way would a Point system solve this? Since for a regular person, points wouldn't give any value either, right?
subscription on the other hand could remove ads which has a value.

---

Another problem I want to note is that users may not be the best at deciding who gets paid how much. 

People would properly only donate to the biggest content mods that change the most for them.
this would 

1. concentrate the money on few mods
2. let "support" mods fall through complitly e.g. ETF, EMF, architectury-api, ... these mods are essential but don't add any content for players directly

note: Paying every mod equally like a subscription would do isn't perfect either since this would discourage big content mods, and encourage spamming libraries

tbh, I don't know a good solution for this since neither a modrinth subscription nor points would be perfect, maybe a combination? 
just a problem I wanted to talk about once.

**Reply by brawaru** [reply]
> maybe a combination?

That's exactly the proposal. You can subscribe to Modrinth, and you can also buy and use points. They are not mutually exclusive ways of earning revenue. In fact, I noticed that I accidentally deleted the point for potential of integrating subscription and points while editing the draft, I added it back now:

> (Area to explore:) Discount to buying points by subscribing to Modrinth or free points to spend every month of subscription.

---

> concentrate the money on few mods

> let "support" mods fall through complitly e.g. ETF, EMF, architectury-api, ... these mods are essential but don't add any content for players directly

These are completely different concerns, and it applies even to existing ad revenue since downloads give almost no value at the moment and nobody looks at libraries project pages, launchers auto-download dependencies for you. Possible solution would be to allow creators to learn of projects they depend on and support them. I don't think it's in the scope of this proposal.

**Comment by neirenoir** [comment]
While I agree that mods should probably be able to move more money, maybe paid mods or donations are not the best way to go about it for different reasons. First and foremost, the most obvious reason is the EULA prohibits it, but even if it was allowed, the community absolutely abhors paid mods ~~and paying in general, if they can avoid it~~. The backlash would be insane if Modrinth even dared to approach the idea judging by the the whole Jet fiasco, who was essentially a rando in the large scheme of things. Now, everyone is fine with donations, because they can simply skip them. Of course, that leads to the ugly truth of some of the largest modders being homeless. Donations are nice, but they are a drop in the bucket and can't be relied on, especially when it comes to the sustained health of a platform, let alone modders themselves.

Now, the thing is many modders do it for free because we like it. I would like it more if I could earn some extra bucks with my hobby, but that's besides the point. I will admit it does feel kinda unjust when YouTubers making videos about mods probably make more money than the devs making those mods will ever see, and this is especially true for "invisible" mods, like performance mods, which are used by nearly every single serious Minecraft player (includes YouTubers), but rarely mention it; but that, as well, is besides the point. The point is Modrinth needs a steady cashflow just to pay for CDN, servers and employees, and then paying modders with what's left. If the cashflow is negative, then Modrinth will eventually stop existing period. I do not want Modrinth to vanish because I love the ethos of everything it stands for, so I hope everyone reading this realizes that Modrinth *needs* to make money, and the more the better, as it would allow the platform to grow. How the platform makes money is important, though, which is why I propose the following ethical ways to increase revenue.

## Modrinth Forums
I don't think I am the only one who finds the lack of modern Minecraft modding forums troublesome. Ever since Discord got popular, everyone dropped out of more traditional message boards in favour of isolated island chats that are indexed nowhere. I assume that's the reason most of the modding questions you find on Google are from older versions. I think there is a real need for a modding forum where users can discuss mods, modding and mod development.

I know Modrinth is understaffed, and even the mod comments feature is kind of on hold due to lack of moderators, but managing a forum should be easier to manage than many isolated discussions as it would happen with mod comment sections. Better indexing and hyperlinking means more eyes, and more eyes also mean more reports from passerbies.

"But how does this increase revenue?". Well, the number one income source of Modrinth is ads. Ads generate revenue when they are displayed to visitors (I guess Adrinth does it this way too; every ad platform I know does it this way). Users staying on the website means more page impressions, which in turn means more ads being printed, which translates to more money for Modrinth. This is why providing fun activities for users is important, because otherwise people will understand Modrinth as a mere CDN rather than an actual platform. Heck, modpack users do not even have to engage with the website if they use alternative launchers like Prism, so having something else to do than just viewing new mods and downloading mods is good for the website.

## Server hosting and friends list
*I've been told this is a bad idea because it is too complicated to implement, but I don't think there is free lunch to be had. If venturing beyond mod distribution is too much, then I guess we could partner with a server provider like BisectHosting to act as a bulk reseller.*

But anyway, why would we want that? Well, hosting a modded server is, simply put, kind of a pain in the ass. Servers are tedious to set up and maintain, and modded servers even more so. As far as I know, there is no server host that provides an easy to use Minecraft interface for mod and mod config management. How cool could it be to set up a server with a modpack and having it run live for your friends in a few clicks? Oh, yeah, your friends. You gotta explain to them how to set up modded Minecraft. It's not hard, but a lot of my friends had their brains unplug while explaining it. If Modrinth had a friends list, they could simply click on me, and then on "join server" to automatically handle the download of the required mods before launching the relevant instance. You could even have a Modrinth servers list that lets you do this as well. How cool is that!?

If Modrinth had a subscription service, part of the subscription could be destined to paying for the server owners (which could be yourself, essentially paying off your own Minecraft server). If several subscribed friends join, the costs can be shared, and the rest goes to the modders that have mods running in that server. If punctual payments or a points system are introduced, server owners could set up cosmetics shops through the Modrinth app itself, further increasing volume.

## Rich Modrinth presence mod
Building on the idea of a friends list, Modrinth could provide a mod that is recommended by default in new modpacks being built through the app or web. This mod could provide extra features related to Modrinth, like talking to friends or friend groups through Minecraft whispers, or ~~"ugh", I know~~ cosmetics like Essentials. Cosmetic shops in-game are icky, but people don't seem to mind as much when they are in the launcher instead, as shown by the Steam market. This would essentially be the Steam overlay for Minecraft.

## Modder tools
One of the things Modrinth does is that it does not let you include mods from sites other than Modrinth or Github. That does make sense. The thing is, a lot of modders from that other modding website use licenses that do not allow their modpack to be sourced from places they don't control, so compiling and redistributing the .jar as a blob is a no-no. However... what if Modrinth subscribers were allotted some resources so they could compile mods directly from GitHub, keep the GitHub user ID of the repo owner, and essentially make a "virtual account" for these modders in Modrinth? They could accumulate money from their downloads made through their repo, while still technically sourcing from their repo. All they have to do is make an account at Modrinth to claim the money, and there could be a beneficial conversion rate in the process.

**Reply by piprett** [reply]
Regarding your forums idea, there is a reason forums are dying. Your average modding joe prefers to post to discord. The barrier to entry is much lower, as they already have an account, and it's significantly less formal.

I agree that server hosting is interesting, however I don't believe we should go down a venture of friend lists, server lists, etc. I also don't believe we should go into hosting server shops for "cosmetics".

I don't like the idea of Modrinth cosmetics, and I again don't like the friend list idea. Modrinth should do one thing, and do it well. That thing is distributing game modifications.

You want to compile & distribute jars from GitHub, which I also don't like. The author themself should be the one wanting to add their mods to Modrinth. Then taking their revenue, that just feels unethical.

**Reply by neirenoir** [reply]
>Your average modding joe prefers to post to discord. The barrier to entry is much lower, as they already have an account, and it's significantly less formal.

To be fair, if they are modders that have already uploaded something to Modrinth, they already have an account.

>Modrinth should do one thing, and do it well. That thing is distributing game modifications.

The Unix philosophy is nice and all until you need to start thinking about making money. The dev of GPG was broke for a long time until he mentioned he may have to discontinue the software to dedicate himself to something else. Being a CDN does not pay well unless you provide to third parties, which is not the case with Modrinth.

>You want to compile & distribute jars from GitHub, which I also don't like. The author themself should be the one wanting to add their mods to Modrinth. Then taking their revenue, that just feels unethical.

I mean, the alternative is people uploading their jars as blobs without linking back to the original author. I am not sure the Modrinth modpack review process checks for licenses of blobs, especially when sometimes the licensing clauses of these mods are obtuse enough to require a lawyer.

**Reply by Minenash** [reply]
> To be fair, if they are modders that have already uploaded something to Modrinth, they already have an account.

But wouldn't most forum users be the users of the mods rather then the devs?

**Reply by neirenoir** [reply]
> But wouldn't most forum users be the users of the mods rather then the devs?

Ideally, both devs and users would use the forum, but yeah. You can increase ad revenue merely by increasing engagement of mod developers, but if it attracts users it would be massively useful for everyone involved, considering it is rumored only logged in users contribute to developer ad revenue (maybe just an implementation limitation, tho; I assume it has to do with caches). You can sign in with a gazillion different OAuth providers, tho, so registering is already kind of low friction.
